units/time_series_update.go

```diff
@@ -81,75 +91,45 @@ func (j *timeSeriesUpdateJob) Run(ctx context.Context) {
 		}
 	}
 
+	baseID := j.Series.CreateBaseSeriesID()
+
 	if j.conf.Flags.DisableSignalProcessing {
-		grip.InfoWhen(sometimes.Percent(10), j.makeMessage("signal processing is disabled, skipping processing", j.PerformanceResultId))
+		grip.InfoWhen(sometimes.Percent(10), j.makeMessage("signal processing is disabled, skipping processing"))
 		return
 	}
 
-	if j.performanceAnalysisService == nil {
-		j.performanceAnalysisService = perf.NewPerformanceAnalysisService(j.conf.ChangeDetector.URI, j.conf.ChangeDetector.User, j.conf.ChangeDetector.Token)
+	if j.service == nil {
+		j.service = perf.NewPerformanceAnalysisService(j.conf.ChangeDetector.URI, j.conf.ChangeDetector.User, j.conf.ChangeDetector.Token)
 	}
 
-	performanceData, err := model.GetPerformanceData(ctx, j.env, j.PerformanceResultId)
-	if err != nil {
-		j.AddError(errors.Wrapf(err, "aggregating time perfData %s", j.PerformanceResultId.String()))
-		return
-	}
-	if performanceData == nil {
-		j.AddError(model.MarkPerformanceResultsAsAnalyzed(ctx, j.env, j.PerformanceResultId))
-		return
-	}
-	for _, perfData := range performanceData {
-		sort.Slice(perfData.TimeSeries, func(i, j int) bool {
-			return perfData.TimeSeries[i].Order < perfData.TimeSeries[j].Order
-		})
-		series := perf.TimeSeriesModel{
-			Project:     perfData.PerformanceResultId.Project,
-			Variant:     perfData.PerformanceResultId.Variant,
-			Task:        perfData.PerformanceResultId.Task,
-			Test:        perfData.PerformanceResultId.Test,
-			Measurement: perfData.PerformanceResultId.Measurement,
-		}
-		series.Arguments = []perf.ArgumentsModel{}
-		for k, v := range perfData.PerformanceResultId.Arguments {
-			series.Arguments = append(series.Arguments, perf.ArgumentsModel{
-				Name:  k,
-				Value: v,
-			})
-		}
-		filteredData := filterOldExecutions(perfData.TimeSeries)
```

Comment:
is the idea that we no longer need this because the stat names are the same for every execution of the task and they're only going to fetch the values from the latest executions of each task?

Reply:
We can remove this because it is now done on the SPS side—they take care of filtering out older executions.